### PR TITLE
Update to latest code base from Microsoft

### PR DESCRIPTION
**Pull request by dclaux**
This updates the API to its latest code base from Microsoft. Thanks to @LukeMarlin and @rich-perry for their contributions.

Notable changes:
- Removed ResourceManager and need for resx file
- Added missing values in WellKnownFolderName enumeration
- Added support for GetUserUnifiedGroups
- Added support for GetUserPhoto
- Added support for FindPeople
- Minor bug fixes

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@dclaux__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>You've already signed the contribution license agreement. Thanks!</span>
        <span>We will now validate the agreement and then real humans will evaluate your PR.</span>

TTYL, AZPRBOT;

File: .gitignore [hunk 1]
```diff
@@ -1,184 +0,0 @@
-## Ignore Visual Studio temporary files, build results, and
```
Comment (vboctor) [inline .gitignore]: We shouldn't be deleting the .gitignore file.  Please re-add it.

File: ComplexProperties/AttributedString.cs [hunk 2]
```diff
@@ -0,0 +1,137 @@
+// ---------------------------------------------------------------------------
```
Comment (vboctor) [inline ComplexProperties/AttributedString.cs]: Possibly in a separate checkin, but we should revise all the headers to reflect the MIT license.  Ping me offline if you want to agree on the content before you take the time to apply to all files.

File: README.md [hunk 3]
```diff
@@ -1,16 +1,15 @@
 # Getting Started with the EWS Managed API
-[![Gitter](https://badges.gitter.im/Join Chat.svg)](https://gitter.im/OfficeDev/ews-managed-api?utm_source=badge&utm_medium=badge&utm_campaign=pr-badge&utm_content=badge)
```
Comment (vboctor) [inline README.md]: Why did you remove the link to the chat room?

Comment (vboctor) [inline README.md]: It is likely that the readme on github is newer than the one we had internally.

**Comment by vboctor** [comment]
Few high level comments:

- Fixing the headers to reference MIT license can be a separate follow up pull request.  I think we did that on the Java API, you may want to start with that.
- The description of the pull request/commit should include the descriptions of the checkins that we ported across, so we have a record of what new features / bug fixes were brought over.  Similar to how you mentioned the WellKnownFolder point.
- Did we include the WellKnownFolder fix as part of this or will it be in a separate checkin?
- .gitignore should be re-added and readme diff should be closely reviewed.

**Comment by dclaux** [comment]
@vboctor
Restored .gitignore and README.md
Added description of changes in pull request comment
Will update headers in a separate pull request

**Comment by vboctor** [comment]
I've squashed this into a single checkin with the latest description and merged it via 45459165001657f3db7d92d094b960d3abec390f.
